test/checkChallenge.test.js

```diff
@@ -286,6 +286,111 @@ describe('Challenges', () => {
    
   });
 
+  describe('Rejectnew', () => {
+
+    it('reject all games if rejectnew is used with default rejectnewmsg', () => {
+
+      const notification = base_challenge({ ranked: false });
+
+      config.rejectnew = true;
+      
+      const result = conn.checkChallengeBot(notification);
+      
+      assert.deepEqual(result, ({ reject: true, msg: 'Currently, this bot is not accepting games, try again later' }));
+    });
+
+    // rejectnew is ranked-unranked aspecific, so ranked unranked testing is minimal
+    it('also rejectnew in ranked games', () => {
+
+        const notification = base_challenge({ ranked: true });
+  
+        config.rejectnew = true;
+        
+        const result = conn.checkChallengeBot(notification);
+        
+        assert.deepEqual(result, ({ reject: true, msg: 'Currently, this bot is not accepting games, try again later' }));
+    });
+
+    it('reject all games if rejectnew is used with custom rejectnewmsg', () => {
+
+        const notification = base_challenge({ ranked: false });
+  
+        config.rejectnew = true;
+        config.rejectnewmsg = 'Sorry, i am not available now.';
+        
+        const result = conn.checkChallengeBot(notification);
+        
+        assert.deepEqual(result, ({ reject: true, msg: 'Sorry, i am not available now.' }));
+    });
+
+    it('do not reject game if rejectnew is not used, and rejectnewmsg is default', () => {
+
+        const notification = base_challenge({ ranked: false });
+        
+        const result = conn.checkChallengeBot(notification);
+        
+        assert.deepEqual(result, ({ reject: false }));
+    });
+
+    it('also do not reject game in ranked games', () => {
+
+        const notification = base_challenge({ ranked: true });
+        
+        const result = conn.checkChallengeBot(notification);
+        
+        assert.deepEqual(result, ({ reject: false }));
+    });
+
+    it('do not reject game if rejectnew is not used, and rejectnewmsg is custom', () => {
+
+        const notification = base_challenge({ ranked: false });
+
+        config.rejectnewmsg = 'Sorry, i am not available now.';
+        
+        const result = conn.checkChallengeBot(notification);
+        
+        assert.deepEqual(result, ({ reject: false }));
+    });
+
+    it('reject all games if rejectnewfile is used and a reject file exists', () => {
```

Comment:
Shouldn't we mock fs?

Reply:
Actually i'm not too sure, as long as we can cleanly refer to the right file and have no side-effect i think it's kinda ok to test like this.

Reply:
oh interesting i didnt think at all to write testing for `fs`

i read this and it seems interesting:
https://github.com/tschaub/mock-fs

i'll try to rewrite a mocked-fs version of the test